python/pyarrow/tests/test_dataset.py

```diff
@@ -16,31 +16,29 @@
 # under the License.
 
 import contextlib
-import os
-import posixpath
```

Comment:
This probably went wrong in the rebase, you accidentally removed `posixpath`, which now causes some of the CI failures